kubeflow/components/generate-dataset.yaml

```diff
@@ -0,0 +1,48 @@
+name: Generate dataset
+description: |
+  Generate dataset
+inputs:
+  - {name: image, description: '', default: ''}
+  - {name: s3_bucket, description: '', default: ''}
+  - {name: owner, description: '', default: ''}
+  - {name: project, description: '', default: ''}
+  - {name: experiment, description: '', default: ''}
+  - {name: dataset, description: '', default: ''}
+  - {name: parallel, description: '', default: ''}
+  - {name: additional_args, description: '', default: ''}
+outputs:
+  - {name: dataset_path}
+implementation: 
+  container:
+    image: '{{inputs.parameters.image}}'
+    command:
+    - /bin/bash
+    - -ex
+    - -c
+    - |
+      . /opt/genie-toolkit/lib.sh
+      parse_args "$0" "image s3_bucket owner project experiment dataset parallel dataset_path" "$@"
```

Comment:
It would be nice to have a "genie_version" parameter here, and fetch/checkout/install the right version of genie-toolkit as first step. Same for "thingtalk_version", and "genienlp_version" for the train step.